fs/romfs/fs_romfsutil.c

```diff
@@ -436,51 +429,44 @@ static int romfs_cachenode(FAR struct romfs_mountpt_s *rm,
           return ret;
         }
 
-      /* Now we are pointing to the real entry of interest. Is it a
-       * directory? Or a file?
-       */
+      ret = romfs_parsefilename(rm, offset, childname);
+      if (ret < 0)
+        {
+          return ret;
+        }
 
-      if (IS_DIRECTORY(next) || IS_FILE(next))
+      if (strcmp(childname, ".") != 0 && strcmp(childname, "..") != 0)
         {
-          ret = romfs_parsefilename(rm, offset, childname);
-          if (ret < 0)
+          if (child == NULL || nodeinfo->rn_count == num - 1)
             {
-              return ret;
-            }
+              FAR void *tmp;
 
-          if (strcmp(childname, ".") != 0 && strcmp(childname, "..") != 0)
-            {
-              if (child == NULL || nodeinfo->rn_count == num - 1)
+              tmp = kmm_realloc(nodeinfo->rn_child,
+                                (num + NODEINFO_NINCR) *
+                                sizeof(FAR struct romfs_nodeinfo_s *));
```

Comment:
```suggestion
                                sizeof(struct romfs_nodeinfo_s *));
```
actually we can replace it even with `void *`, but change to be `sizeof(*nodeinfo->rn_child)` seems to be even more reasonable.

Reply:
I'm fine just to remove `FAR` from `sizeof` and I can do other changes in a separate PR

Reply:
Done

Reply:
but FAR is required here, otherwise the sizeof may return value smaller than expected.